MatrixMultiplication/MatrixMultiplication/Matrix.cs

```diff
@@ -0,0 +1,245 @@
+using System.Text;
+
+namespace MatrixMultiplication;
+
+public class Matrix
+{
+    private readonly int[,] _matrix;
+    
+    /// <summary>
+    /// Gets the number of rows in the matrix.
+    /// </summary>
+    public int Rows => _matrix.GetLength(0);
+    
+    /// <summary>
+    /// Gets the number of columns in the matrix.
+    /// </summary>
+    public int Columns => _matrix.GetLength(1);
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Matrix"/> class.
+    /// </summary>
+    /// <param name="data">The 2D array representing the matrix elements.</param>
+    public Matrix(int[,] data)
+    {
+        this._matrix = (int[,])(data.Clone());
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Matrix"/> class by loading data from file.
+    /// </summary>
+    /// <param name="filePatch">The file path containing the matrix data.</param>
+    public Matrix(string filePatch)
+    {
+        if (!File.Exists(filePatch))
+        {
+            throw new FileNotFoundException(filePatch);
+        }
+        
+        var lines = File.ReadAllLines(filePatch);
+
+        if (lines.Length == 0)
+        {
+            throw new ArgumentException("File is empty.");
+        }
+        
+        _matrix = new int[lines.Length, lines[0].Split(' ').Length];
+        
+        for (var i = 0; i < lines.Length; i++)
+        {
+            if (lines[i] == string.Empty)
+            {
+                throw new ArgumentException("Empty row detected");
+            }
+            
+            var line = lines[i].Split(' ').Select(int.Parse).ToArray();
+
+            if (line.Length != lines.Length)
+            {
+                throw new ArgumentException("Matrix is not rectangular");
+            }
+            
+            for (var j = 0; j < line.Length; j++)
+            {
+                _matrix[i, j] = line[j];
+            }
+        }
+        
+    }
+    /// <summary>
+    /// Generates a matrix with random integer values.
+    /// </summary>
+    /// <param name="rows">The number of rows in the matrix.</param>
+    /// <param name="columns">The number of columns in the matrix.</param>
+    public static Matrix GenerateMatrix(int rows, int columns)
+    {
+        var data = new int[rows, columns];
+        
+        var random = new Random();
+        
+        for (var i = 0; i < rows; i++)
+        {
+            for (var j = 0; j < columns; j++)
+            {
+                data[i, j] = random.Next(0, 16);
+            }
+        }
+        
+        return new Matrix(data);
+    }
+    /// <summary>
```

Comment:
```suggestion
    }

    /// <summary>
```
Тут и ниже, по стайлгайду надо пустую строку между методами